src/content/tutorial/react-step-3/react-step-3.mdx

```diff
@@ -301,233 +266,188 @@ const getRowItems = rows =>
-                  </TableHeader>
-                ))}
-              </TableRow>
-            </TableHead>
-            <TableBody>
-              {rows.map(row => (
-                <React.Fragment key={row.id}>
-                  <TableExpandRow {...getRowProps({ row })}>
-                    {row.cells.map(cell => (
-                      <TableCell key={cell.id}>{cell.value}</TableCell>
-                    ))}
-                  </TableExpandRow>
-                  <TableExpandedRow colSpan={headers.length + 1}>
-                    <h1>Expandable row content</h1>
-                    <p>Description here</p>
-                  </TableExpandedRow>
-                </React.Fragment>
-              ))}
-            </TableBody>
-          </Table>
-        </TableContainer>
-      )}
     />
   );
-};
 ```
 
-We now need to wire up the data table to the render props. First, replace `initialRows` with `rows` now that the rows are coming in from props.
+We need to tell the loading skeleton how many rows to render, so let's use 10 skeleton rows to prepare for the next enhancement...
 
-Delete the `<h1>Expandable row content</h1>` and modify the paragraph element below it to use the `getRowDescription` function.
+## Add pagination
 
-```html
-<p className="description">{getRowDescription(row.id)}</p>
+Pagination! Instead of rendering every repository, let's add pagination to the data table to only render 10 at a time. Depending on your specific requirements, you may need to fetch new data each time that you interact with the pagination component, but for simplicity, we're going to make one request to fetch all data, and then paginate the in-memory row data.
+
+We'll be using [React Hooks](https://reactjs.org/docs/hooks-intro.html) to manage our state. Hooks are a relatively new React API that allows us to author a React component's stateful logic in a function component rather than a class component. Using hooks means we don't need to worry about complex lifecycle methods.
+
+Import React's [useState](https://reactjs.org/docs/hooks-state.html) by modifying the `React` import.
+
+##### src/content/RepoPage/RepoPage.js
+
+```javascript
+import React, { useState } from 'react';
+```
+
+Then initialize the new state variables that we'll use for pagination as the first lines inside the `RepoPage` component, above the `return()`.
+
+##### src/content/RepoPage/RepoPage.js
+
+```javascript
+const [totalItems, setTotalItems] = useState(0);
+const [firstRowIndex, setFirstRowIndex] = useState(0);
+const [currentPageSize, setCurrentPageSize] = useState(10);
 ```
 
-_Note: The `getTableProps`, `getHeaderProps`, and `getRowProps` functions each tell the DataTable component to process and return the necessary props that are needed by the Table, Header and Row components respectively. We take the return of that function (an object containing the props) and [spread it out](https://reactjs.org/docs/jsx-in-depth.html#spread-attributes) into the component._.
```

Comment:
This is also critical info was it moved?

Reply:
It should have been. I've made sure this is included when data table is first used in the previous step.